src/components/error-message/ErrorMessage.tsx

```diff
@@ -1,23 +1,37 @@
-import React, { HTMLProps } from 'react';
+import React, { HTMLProps, useContext, useEffect } from 'react';
+
+// eslint-disable-next-line import/no-named-as-default
+import FormGroupContext from '../formgroup/FormGroupContext';
+// eslint-disable-next-line
 import classNames from 'classnames';
 
 export interface ErrorMessageProps extends HTMLProps<HTMLSpanElement> {
   visuallyHiddenText?: false | string;
 }
 
-const ErrorMessage: React.FC<ErrorMessageProps> = ({
-  className,
-  visuallyHiddenText,
-  children,
-  ...rest
-}) => (
-  <span className={classNames('nhsuk-error-message', className)} {...rest}>
-    {visuallyHiddenText !== false ? (
-      <span className="nhsuk-u-visually-hidden">{visuallyHiddenText}</span>
-    ) : null}
-    {children}
-  </span>
-);
+const ErrorMessage: React.FC<ErrorMessageProps> = props => {
+  const { id, className, visuallyHiddenText, children, ...rest } = props;
+  const { isInFormGroup, inputID, setError } = useContext(FormGroupContext);
+  useEffect(() => {
```

Comment:
This could be simplified down to:

```js
useEffect(() => {
    setError(isInFormGroup || inputID);
    return () => setError(false);
}, [isInFormGroup, inputID]);
```

Reply:
Resolved